### ISSUE TITLE
Send the "father" node as a parameter of bfs 

### ISSUE THREAD

**Issue by yuvaleliav** [graphology/graphology#376]
I want to use the bfsFromNode function of the graphology-traversal package, but I need to know the node that added the current node to the queue. 
Sure I can just keep a map of < NodeId : Depth > inside the bfs, and pick the one that sent it to the current node from the minimal depth of the neighbors of the current node, but it would be much simpler to send it as a parameter of the bfs.

**Comment by Yomguithereal** [comment]
@yuvaleliav can you describe your use case please? I may also add a version of the traversal yielding the path history and this can fit your need also but it might be a little less performant. Yielding the parent could have the same intended use so I just want to know the scenario your are encountering for better designing this feature.

**Comment by yuvaleliav** [comment]
Okay, each node has a finish and start date, for 2 nodes that are connected with an edge the time cannot overlap, when a change is made to the graph, I want to traverse with bfs (from the point of change) and fix the graph if needed.
To see if I need to make a change i can compare the current item to the "parent" (because if the current node needs fixing is relative to the parent node)
or I could check all the neighbors and fix according to all of them, only the parent will need changing as the graph is assumed to be correct because of the change
basically in my use case it is much more intuitive to send the parent as a parameter rather than try to understand who my parent was after currently using a specific node.

**Comment by yuvaleliav** [comment]
Okay I just realized what I need is topological sort and not bfs, probably no need for this feature then

**Comment by Yomguithereal** [comment]
I guess you can add another issue to add topological sort to `graphology-dag` then :)
